PullDependencies does not support http proxy.

Druid source code is not able to be built successfully when http proxy must be used.
The error is like following:
```
2018-03-01T17:50:59,660 ERROR [main] io.druid.cli.PullDependencies - Unable to resolve artifacts for [io.druid.extensions:druid-hdfs-storage:jar:0.13.0-SNAPSHOT (runtime) -> [] < [ (https://repo1.maven.org/maven2/, releases+snapshots)]].
org.eclipse.aether.resolution.DependencyResolutionException: Failed to collect dependencies at io.druid.extensions:druid-hdfs-storage:jar:0.13.0-SNAPSHOT -> org.apache.hadoop:hadoop-client:jar:2.7.3
        at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:380) ~[aether-impl-0.9.0.M2.jar:?]
        at io.tesla.aether.internal.DefaultTeslaAether.resolveArtifacts(DefaultTeslaAether.java:289) ~[tesla-aether-0.0.5.jar:0.0.5]
        at io.druid.cli.PullDependencies.downloadExtension(PullDependencies.java:399) [druid-services-0.13.0-SNAPSHOT.jar:0.13.0-SNAPSHOT]
        at io.druid.cli.PullDependencies.run(PullDependencies.java:298) [druid-services-0.13.0-SNAPSHOT.jar:0.13.0-SNAPSHOT]
        at io.druid.cli.Main.main(Main.java:116) [druid-services-0.13.0-SNAPSHOT.jar:0.13.0-SNAPSHOT]
Caused by: org.eclipse.aether.collection.DependencyCollectionException: Failed to collect dependencies at io.druid.extensions:druid-hdfs-storage:jar:0.13.0-SNAPSHOT -> org.apache.hadoop:hadoop-client:jar:2.7.3
        at org.eclipse.aether.internal.impl.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:292) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:342) ~[aether-impl-0.9.0.M2.jar:?]
        ... 4 more
Caused by: org.eclipse.aether.resolution.ArtifactDescriptorException: Failed to read artifact descriptor for org.apache.hadoop:hadoop-client:jar:2.7.3
        at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:335) ~[maven-aether-provider-3.1.1.jar:3.1.1]
        at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:217) ~[maven-aether-provider-3.1.1.jar:3.1.1]
        at org.eclipse.aether.internal.impl.DefaultDependencyCollector.process(DefaultDependencyCollector.java:461) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:261) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:342) ~[aether-impl-0.9.0.M2.jar:?]
        ... 4 more
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not transfer artifact org.apache.hadoop:hadoop-client:pom:2.7.3 from/to  (https://repo1.maven.org/maven2/): connect timed out
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:459) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:262) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:239) ~[aether-impl-0.9.0.M2.jar:?]
        at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:320) ~[maven-aether-provider-3.1.1.jar:3.1.1]
        at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:217) ~[maven-aether-provider-3.1.1.jar:3.1.1]
        at org.eclipse.aether.internal.impl.DefaultDependencyCollector.process(DefaultDependencyCollector.java:461) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:261) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:342) ~[aether-impl-0.9.0.M2.jar:?]
        ... 4 more
Caused by: org.eclipse.aether.transfer.ArtifactTransferException: Could not transfer artifact org.apache.hadoop:hadoop-client:pom:2.7.3 from/to  (https://repo1.maven.org/maven2/): connect timed out
        at io.tesla.aether.connector.AetherRepositoryConnector$2.wrap(AetherRepositoryConnector.java:830) ~[aether-connector-okhttp-0.0.9.jar:0.0.9]
        at io.tesla.aether.connector.AetherRepositoryConnector$2.wrap(AetherRepositoryConnector.java:824) ~[aether-connector-okhttp-0.0.9.jar:0.0.9]
        at io.tesla.aether.connector.AetherRepositoryConnector$GetTask.flush(AetherRepositoryConnector.java:619) ~[aether-connector-okhttp-0.0.9.jar:0.0.9]
        at io.tesla.aether.connector.AetherRepositoryConnector.get(AetherRepositoryConnector.java:238) ~[aether-connector-okhttp-0.0.9.jar:0.0.9]
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:535) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:436) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:262) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:239) ~[aether-impl-0.9.0.M2.jar:?]
        at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:320) ~[maven-aether-provider-3.1.1.jar:3.1.1]
        at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:217) ~[maven-aether-provider-3.1.1.jar:3.1.1]
        at org.eclipse.aether.internal.impl.DefaultDependencyCollector.process(DefaultDependencyCollector.java:461) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:261) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:342) ~[aether-impl-0.9.0.M2.jar:?]
        ... 4 more
Caused by: java.net.SocketTimeoutException: connect timed out
        at java.net.PlainSocketImpl.socketConnect(Native Method) ~[?:1.8.0_151]
        at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350) ~[?:1.8.0_151]
        at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206) ~[?:1.8.0_151]
        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188) ~[?:1.8.0_151]
        at java.net.Socket.connect(Socket.java:589) ~[?:1.8.0_151]
        at com.squareup.okhttp.Connection.connect(Connection.java:100) ~[okhttp-1.0.2.jar:?]
        at com.squareup.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:287) ~[okhttp-1.0.2.jar:?]
        at com.squareup.okhttp.internal.http.HttpEngine.sendSocketRequest(HttpEngine.java:248) ~[okhttp-1.0.2.jar:?]
        at com.squareup.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:197) ~[okhttp-1.0.2.jar:?]
        at com.squareup.okhttp.internal.http.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:388) ~[okhttp-1.0.2.jar:?]
        at com.squareup.okhttp.internal.http.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:339) ~[okhttp-1.0.2.jar:?]
        at com.squareup.okhttp.internal.http.HttpURLConnectionImpl.getResponseCode(HttpURLConnectionImpl.java:540) ~[okhttp-1.0.2.jar:?]
        at com.squareup.okhttp.internal.http.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:148) ~[okhttp-1.0.2.jar:?]
        at io.tesla.aether.okhttp.OkHttpAetherClient$ResponseAdapter.getStatusCode(OkHttpAetherClient.java:196) ~[aether-connector-okhttp-0.0.9.jar:0.0.9]
        at io.tesla.aether.connector.AetherRepositoryConnector$GetTask.resumableGet(AetherRepositoryConnector.java:549) ~[aether-connector-okhttp-0.0.9.jar:0.0.9]
        at io.tesla.aether.connector.AetherRepositoryConnector$GetTask.run(AetherRepositoryConnector.java:391) ~[aether-connector-okhttp-0.0.9.jar:0.0.9]
        at io.tesla.aether.connector.AetherRepositoryConnector.get(AetherRepositoryConnector.java:232) ~[aether-connector-okhttp-0.0.9.jar:0.0.9]
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:535) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:436) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:262) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:239) ~[aether-impl-0.9.0.M2.jar:?]
        at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:320) ~[maven-aether-provider-3.1.1.jar:3.1.1]
        at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:217) ~[maven-aether-provider-3.1.1.jar:3.1.1]
        at org.eclipse.aether.internal.impl.DefaultDependencyCollector.process(DefaultDependencyCollector.java:461) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:261) ~[aether-impl-0.9.0.M2.jar:?]
        at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:342) ~[aether-impl-0.9.0.M2.jar:?]
        ... 4 more
Exception in thread "main" java.lang.RuntimeException: org.eclipse.aether.resolution.DependencyResolutionException: Failed to collect dependencies at io.druid.extensions:druid-hdfs-storage:jar:0.13.0-SNAPSHOT -> org.apache.hadoop:hadoop-client:jar:2.7.3
        at com.google.common.base.Throwables.propagate(Throwables.java:160)
        at io.druid.cli.PullDependencies.downloadExtension(PullDependencies.java:412)
        at io.druid.cli.PullDependencies.run(PullDependencies.java:298)
        at io.druid.cli.Main.main(Main.java:116)
Caused by: org.eclipse.aether.resolution.DependencyResolutionException: Failed to collect dependencies at io.druid.extensions:druid-hdfs-storage:jar:0.13.0-SNAPSHOT -> org.apache.hadoop:hadoop-client:jar:2.7.3
        at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:380)
        at io.tesla.aether.internal.DefaultTeslaAether.resolveArtifacts(DefaultTeslaAether.java:289)
        at io.druid.cli.PullDependencies.downloadExtension(PullDependencies.java:399)
        ... 2 more
Caused by: org.eclipse.aether.collection.DependencyCollectionException: Failed to collect dependencies at io.druid.extensions:druid-hdfs-storage:jar:0.13.0-SNAPSHOT -> org.apache.hadoop:hadoop-client:jar:2.7.3
        at org.eclipse.aether.internal.impl.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:292)
        at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveDependencies(DefaultRepositorySystem.java:342)
        ... 4 more
Caused by: org.eclipse.aether.resolution.ArtifactDescriptorException: Failed to read artifact descriptor for org.apache.hadoop:hadoop-client:jar:2.7.3
        at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:335)
        at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:217)
        at org.eclipse.aether.internal.impl.DefaultDependencyCollector.process(DefaultDependencyCollector.java:461)
        at org.eclipse.aether.internal.impl.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:261)
        ... 5 more
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not transfer artifact org.apache.hadoop:hadoop-client:pom:2.7.3 from/to  (https://repo1.maven.org/maven2/): connect timed out
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:459)
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:262)
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:239)
        at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:320)
        ... 8 more
Caused by: org.eclipse.aether.transfer.ArtifactTransferException: Could not transfer artifact org.apache.hadoop:hadoop-client:pom:2.7.3 from/to  (https://repo1.maven.org/maven2/): connect timed out
        at io.tesla.aether.connector.AetherRepositoryConnector$2.wrap(AetherRepositoryConnector.java:830)
        at io.tesla.aether.connector.AetherRepositoryConnector$2.wrap(AetherRepositoryConnector.java:824)
        at io.tesla.aether.connector.AetherRepositoryConnector$GetTask.flush(AetherRepositoryConnector.java:619)
        at io.tesla.aether.connector.AetherRepositoryConnector.get(AetherRepositoryConnector.java:238)
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:535)
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:436)
        ... 11 more
Caused by: java.net.SocketTimeoutException: connect timed out
        at java.net.PlainSocketImpl.socketConnect(Native Method)
        at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
        at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
        at java.net.Socket.connect(Socket.java:589)
        at com.squareup.okhttp.Connection.connect(Connection.java:100)
        at com.squareup.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:287)
        at com.squareup.okhttp.internal.http.HttpEngine.sendSocketRequest(HttpEngine.java:248)
        at com.squareup.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:197)
        at com.squareup.okhttp.internal.http.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:388)
        at com.squareup.okhttp.internal.http.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:339)
        at com.squareup.okhttp.internal.http.HttpURLConnectionImpl.getResponseCode(HttpURLConnectionImpl.java:540)
        at com.squareup.okhttp.internal.http.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:148)
        at io.tesla.aether.okhttp.OkHttpAetherClient$ResponseAdapter.getStatusCode(OkHttpAetherClient.java:196)
        at io.tesla.aether.connector.AetherRepositoryConnector$GetTask.resumableGet(AetherRepositoryConnector.java:549)
        at io.tesla.aether.connector.AetherRepositoryConnector$GetTask.run(AetherRepositoryConnector.java:391)
        at io.tesla.aether.connector.AetherRepositoryConnector.get(AetherRepositoryConnector.java:232)
        ... 13 more
```

--- a/docs/content/operations/pull-deps.md
+++ b/docs/content/operations/pull-deps.md
@@ -40,6 +40,30 @@ Don't use the default remote repositories, only use the repositories provided di
 
 Version to use for extension coordinate that doesn't have a version information. For example, if extension coordinate is `io.druid.extensions:mysql-metadata-storage`, and default version is `#{DRUIDVERSION}`, then this coordinate will be treated as `io.druid.extensions:mysql-metadata-storage:#{DRUIDVERSION}`
 
+`--use-proxy`
+
+Use http/https proxy to send request to the remote repository servers. `--proxy-host` and `--proxy-port` must be set explicitly if this option is enabled.  
+
+`--proxy-type`
+
+Set the proxy type, Should be either *http* or *https*, default value is *https*.
+
+`--proxy-host`
+
+Set the proxy host. e.g. proxy.com.
+
+`--proxy-port`
+
+Set the proxy port number. e.g. 8080.
+
+`--proxy-username`
+
+Set a username to connect to the proxy, this option is only required if the proxy server uses authentication.
+
+`--proxy-password`
+
+Set a password to connect to the proxy, this option is only required if the proxy server uses authentication.
+
 To run `pull-deps`, you should
 
 1) Specify `druid.extensions.directory` and `druid.extensions.hadoopDependenciesDir`, these two properties tell `pull-deps` where to put extensions. If you don't specify them,  default values will be used, see [Configuration](../configuration/index.html).

--- a/services/src/main/java/io/druid/cli/PullDependencies.java
+++ b/services/src/main/java/io/druid/cli/PullDependencies.java
@@ -19,6 +19,7 @@
 
 package io.druid.cli;
 
+import com.google.common.base.Strings;
 import com.google.common.base.Throwables;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Lists;
@@ -33,17 +34,23 @@ import io.druid.java.util.common.StringUtils;
 import io.druid.java.util.common.logger.Logger;
 import io.tesla.aether.Repository;
 import io.tesla.aether.TeslaAether;
+import io.tesla.aether.guice.RepositorySystemSessionProvider;
 import io.tesla.aether.internal.DefaultTeslaAether;
 import org.apache.commons.io.FileUtils;
+import org.eclipse.aether.RepositorySystemSession;
 import org.eclipse.aether.artifact.Artifact;
 import org.eclipse.aether.artifact.DefaultArtifact;
 import org.eclipse.aether.collection.CollectRequest;
 import org.eclipse.aether.graph.Dependency;
 import org.eclipse.aether.graph.DependencyFilter;
 import org.eclipse.aether.graph.DependencyNode;
+import org.eclipse.aether.repository.Authentication;
+import org.eclipse.aether.repository.Proxy;
+import org.eclipse.aether.repository.RemoteRepository;
 import org.eclipse.aether.resolution.DependencyRequest;
 import org.eclipse.aether.util.artifact.JavaScopes;
 import org.eclipse.aether.util.filter.DependencyFilterUtils;
+import org.eclipse.aether.util.repository.AuthenticationBuilder;
 
 import java.io.File;
 import java.io.IOException;
@@ -54,6 +61,7 @@ import java.net.URI;
 import java.net.URISyntaxException;
 import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 
 @Command(
@@ -195,6 +203,48 @@ public class PullDependencies implements Runnable
       required = false
   )
   public String defaultVersion = PullDependencies.class.getPackage().getImplementationVersion();
+
+  @Option(
+      name = {"--use-proxy"},
+      title = "Use http/https proxy to pull dependencies.",
+      required = false
+  )
+  public boolean useProxy = false;
+
+  @Option(
+      name = {"--proxy-type"},
+      title = "The proxy type, should be either http or https",
+      required = false
+  )
+  public String proxyType = "https";
+
+  @Option(
+      name = {"--proxy-host"},
+      title = "The proxy host",
+      required = false
+  )
+  public String proxyHost = "";
+
+  @Option(
+      name = {"--proxy-port"},
+      title = "The proxy port",
+      required = false
+  )
+  public int proxyPort = -1;
+
+  @Option(
+      name = {"--proxy-username"},
+      title = "The proxy username",
+      required = false
+  )
+  public String proxyUsername = "";
+
+  @Option(
+      name = {"--proxy-password"},
+      title = "The proxy password",
+      required = false
+  )
+  public String proxyPassword = "";
 
   public PullDependencies()
   {
@@ -411,10 +461,7 @@ public class PullDependencies implements Runnable
     }
 
     if (log.isTraceEnabled() || log.isDebugEnabled()) {
-      return new DefaultTeslaAether(
-          localRepository,
-          remoteRepositories.toArray(new Repository[remoteRepositories.size()])
-      );
+      return createTeslaAether(remoteRepositories);
     }
 
     PrintStream oldOut = System.out;
@@ -445,10 +492,7 @@ public class PullDependencies implements Runnable
               StringUtils.UTF8_STRING
           )
       );
-      return new DefaultTeslaAether(
-          localRepository,
-          remoteRepositories.toArray(new Repository[remoteRepositories.size()])
-      );
+      return createTeslaAether(remoteRepositories);
     }
     catch (UnsupportedEncodingException e) {
       // should never happen
@@ -457,6 +501,39 @@ public class PullDependencies implements Runnable
     finally {
       System.setOut(oldOut);
     }
+  }
+
+  private DefaultTeslaAether createTeslaAether(List<Repository> remoteRepositories)
+  {
+    if (!useProxy) {
+      return new DefaultTeslaAether(
+          localRepository,
+          remoteRepositories.toArray(new Repository[remoteRepositories.size()])
+      );
+    }
+
+    if (!StringUtils.toLowerCase(proxyType).equals(Proxy.TYPE_HTTP) && !StringUtils.toLowerCase(proxyType).equals(Proxy.TYPE_HTTPS)) {
+      throw new IllegalArgumentException("invalid proxy type: " + proxyType);
+    }
+
+    RepositorySystemSession repositorySystemSession = new RepositorySystemSessionProvider(new File(localRepository)).get();
+    List<RemoteRepository> rl = remoteRepositories.stream().map(r -> {
+      RemoteRepository.Builder builder = new RemoteRepository.Builder(r.getId(), "default", r.getUrl());
+      if (r.getUsername() != null && r.getPassword() != null) {
+        Authentication auth = new AuthenticationBuilder().addUsername(r.getUsername()).addPassword(r.getPassword()).build();
+        builder.setAuthentication(auth);
+      }
+
+      final Authentication proxyAuth;
+      if (Strings.isNullOrEmpty(proxyUsername)) {
+        proxyAuth = null;
+      } else {
+        proxyAuth = new AuthenticationBuilder().addUsername(proxyUsername).addPassword(proxyPassword).build();
+      }
+      builder.setProxy(new Proxy(proxyType, proxyHost, proxyPort, proxyAuth));
+      return builder.build();
+    }).collect(Collectors.toList());
+    return new DefaultTeslaAether(rl, repositorySystemSession);
   }
 
   /**

